salt/utils/zeromq.py

```diff
@@ -5,21 +5,65 @@
 # Import Python libs
 from __future__ import absolute_import, print_function, unicode_literals
 
-# Import Salt libs
+import logging
+import tornado.ioloop
 from salt.exceptions import SaltSystemExit
 
-# Import 3rd-party libs
+log = logging.getLogger(__name__)
+
 try:
     import zmq
-    HAS_ZMQ = True
 except ImportError:
-    HAS_ZMQ = False
+    zmq = None
+    log.debug('ZMQ module is not found')
+
+ZMQDefaultLoop = None
+ZMQ_VERSION_INFO = (-1, -1, -1)
+
+try:
+    if zmq:
+        ZMQ_VERSION_INFO = [int(v_el) for v_el in zmq.__version__.split('.')]
+        if ZMQ_VERSION_INFO[0] > 16:  # 17.0.x+ deprecates zmq's ioloops
+            ZMQDefaultLoop = tornado.ioloop.IOLoop
+except Exception as ex:
+    log.exception(ex)
```

Comment:
`log.exception()` will already log the exception message itself (including the traceback). Therefore, it's not necessary to capture the exception using `as ex`, and the log message should contain a descriptive message telling what led to the exception being raised. For example:

```python
log.exception('Error encountered doing XXXXX')
```

Reply:
Ahh, right. Thanks! Fixd.